### ISSUE TITLE
Implicit JWT Keys and Verification Methods

### ISSUE THREAD

**Issue by Maho91** [panva/node-oidc-provider#778]
Hi there, 

Can anyone help me with how to configure the RS256 or HS256 keys properly? Because when I use Idtoken of the Implicit Flow the authentication of the users happens and to be logged in to the OP, however, after the login the OP inters in an infinite loop of verification trials of the Idtoken, and it blocks the user from seeing anything. 
Below the can find the error of the OP and the configuration of the IDP

Error
-----------------------------
 Unable to find a suitable verification key for JWS w/ header {"alg":"RS256","typ":"JWT","kid":"r1LkbBo3925Rb2ZFFrKyU3MVex9T2817Kx0vbi6i_Kc"}

My Configuration
-----------------------------

module.exports = {
  clients: [
       {
      client_id: 'some_id',
      client_secret: 'some_secret',
      token_endpoint_auth_method: 'none',
      scope: 'openid email profile',
      grant_types: ['implicit','refresh_token'],
      id_token_signed_response_alg: 'RS256',
      response_types: ['id_token'],
      redirect_uris: ['http://op/callback'],
    }  
],
  interactions: {
    policy: basePolicy,
    url(ctx, interaction) { // eslint-disable-line no-unused-vars
      return `/interaction/${ctx.oidc.uid}`;                                                                                                                                                                                                     },
  },
  cookies: {
    long: { signed: true, maxAge: (1 * 24 * 60 * 60) * 1000 }, // 1 day in ms
    short: { signed: true },
    keys: ['some secret key', 'and also the old rotated away some time ago', 'and one more'],
  },
  claims: {
    email: ['email','mail'],
    phone: ['phone_number'],
    profile: ['birthdate','mail', 'profile', 'email', 'name', 'zoneinfo','domain','id','user','scopedtoken'],
  },
  formats: {
    AccessToken: 'jwt',
    ClientCredentials: 'jwt',
    RefreshToken: 'jwt',
  },
  features: {
    devInteractions: { enabled: false }, // defaults to true
    mTLS: {
      enabled: true,
      selfSignedTlsClientAuth: true,
      getCertificate() {
        return cert;
      },
    },
    deviceFlow: { enabled: true }, // defaults to false
    introspection: { enabled: true }, // defaults to false
    revocation: { enabled: true }, // defaults to false
  },
  jwks: {
    keys: [
      {
        d: 'VEZOsY07JTFzGTqv6cC2Y32vsfChind2I_TTuvV225_-0zrSej3XLRg8iE_u0-3GSgiGi4WImmTwmEgLo4Qp3uEcxCYbt4NMJC7fwT2i3dfRZjtZ4yJwFl0SIj8TgfQ8ptwZbFZUlcHGXZIr4nL8GXyQT0CK8wy4COfmymHrrUoyfZA154ql_OsoiupSUCRcKVvZj2JHL2KILsq_sh_l7g2dqAN8D7jY$        dp: 'E1Y-SN4bQqX7kP-bNgZ_gEv-pixJ5F_EGocHKfS56jtzRqQdTurrk4jIVpI-ZITA88lWAHxjD-OaoJUh9Jupd_lwD5Si80PyVxOMI2xaGQiF0lbKJfD38Sh8frRpgelZVaK_gm834B6SLfxKdNsP04DsJqGKktODF_fZeaGFPH0',                                                           dq: 'F90JPxevQYOlAgEH0TUt1-3_hyxY6cfPRU2HQBaahyWrtCWpaOzenKZnvGFZdg-BuLVKjCchq3G_70OLE-XDP_ol0UTJmDTT-WyuJQdEMpt_WFF9yJGoeIu8yohfeLatU-67ukjghJ0s9CBzNE_LrGEV6Cup3FXywpSYZAV3iqc',
        e: 'AQAB',
        kty: 'RSA',
        n: 'xwQ72P9z9OYshiQ-ntDYaPnnfwG6u9JAdLMZ5o0dmjlcyrvwQRdoFIKPnO65Q8mh6F_LDSxjxa2Yzo_wdjhbPZLjfUJXgCzm54cClXzT5twzo7lzoAfaJlkTsoZc2HFWqmcri0BuzmTFLZx2Q7wYBm0pXHmQKF0V-C1O6NWfd4mfBhbM-I1tHYSpAMgarSm22WDMDx-WWI7TEzy2QhaBVaENW9BKaKkJ$        p: '5wC6nY6Ev5FqcLPCqn9fC6R9KUuBej6NaAVOKW7GXiOJAq2WrileGKfMc9kIny20zW3uWkRLm-O-3Yzze1zFpxmqvsvCxZ5ERVZ6leiNXSu3tez71ZZwp0O9gys4knjrI-9w46l_vFuRtjL6XEeFfHEZFaNJpz-lcnb3w0okrbM',                                                            q: '3I1qeEDslZFB8iNfpKAdWtz_Wzm6-jayT_V6aIvhvMj5mnU-Xpj75zLPQSGa9wunMlOoZW9w1wDO1FVuDhwzeOJaTm-Ds0MezeC4U6nVGyyDHb4CUA3ml2tzt4yLrqGYMT7XbADSvuWYADHw79OFjEi4T3s3tJymhaBvy1ulv8M',
        qi: 'wSbXte9PcPtr788e713KHQ4waE26CzoXx-JNOgN0iqJMN6C4_XJEX-cSvCZDf4rh7xpXN6SGLVd5ibIyDJi7bbi5EQ5AXjazPbLBjRthcGXsIuZ3AtQyR0CEWNSdM7EyM5TRdyZQ9kftfz9nI03guW3iKKASETqX2vh0Z8XRjyU',
        use: 'sig',
      },
    ],
  },
  ttl: {
    AccessToken: 1 * 60 * 60, // 1 hour in seconds
    AuthorizationCode: 10 * 60, // 10 minutes in seconds
    IdToken: 1 * 60 * 60, // 1 hour in seconds
    DeviceCode: 10 * 60, // 10 minutes in seconds
    RefreshToken: 1 * 24 * 60 * 60, // 1 day in seconds
  },
};

--------------------------------------
Thank you in advance

**Comment by panva** [comment]
@Maho91 there's not enough substance in the issue to help out. The calculated thumbprint for the key would be `kiFn67WR2q3aLCOT6OsiD0MuBwcEf9rREzmo57hxrmI`. The error doesn't come from the library either.

**Comment by Maho91** [comment]
Yes, I already mentioned above, that the error comes from the OP because it tries to verify the id token but it cannot find a suitable verification key. Nevertheless, it works properly with Authrozaion_code.

**Comment by panva** [comment]
No, that error does not come from this project (the OP)

**Comment by Maho91** [comment]
Sorry, you are right it is my mistake, I just mixed up between the abbreviations, I meant my other project the relying party (RP). 
I was looking for a way to set up new keys. I find the link (https://mkjwk.org/) to generate JWT  sample keys so I would be able to verify it with this debugger (https://jwt.io/#debugger-io). and I set the private key and the certificate. in the mTLS: {} tag, it worked.

This project is perfect, great job. Thanks for your time, my problem is solved.
